async-book/src/execution/future.md

```diff
@@ -0,0 +1,189 @@
+# The `Future` Trait
+
+The `Future` trait is at the center of asynchronous programming in Rust.
+A `Future` is an asynchronous computation that can produce a value
+(although that value may be empty, e.g. `()`). A *simplified* version of
+the future trait might look something like this:
+
+```rust
+trait SimpleFuture {
+    type Output;
+    fn poll(&mut self, wake: fn()) -> Poll<Self::Output>;
+}
+
+enum Poll<T> {
+    Ready(T),
+    Pending,
+}
+```
+
+Futures can be advanced by calling the `poll` function, which will drive the
+future as far towards completion as possible. If the future completes, it
+returns `Poll::Ready(result)`. If the future is not able to complete yet, it
+returns `Poll::Pending` and arranges for the `wake()` function to be called
+when the `Future` is ready to make more progress. When `wake()` is called, the
+executor driving the `Future` will call `poll` again so that the `Future` can
+make more progress.
+
+Without `wake()`, the executor would have no way of knowing when a particular
+future could make progress, and would have to be constantly polling all every
```

Comment:
`would have to be constantly polling every future`

remove the all after polling

Reply:
Done.